internal/report/json/json.go

```diff
@@ -2,17 +2,23 @@ package json
 
 import (
 	"bytes"
+	"context"
 	"encoding/json"
+	"fmt"
 	"io"
 	"net/http"
+	"net/url"
 	"os"
+	"strings"
 
 	"github.com/rs/zerolog/log"
+	"github.com/saucelabs/saucectl/internal/build"
 	"github.com/saucelabs/saucectl/internal/report"
 )
 
 // Reporter represents struct to report in json format
 type Reporter struct {
+	Service    build.Reader
```

Comment:
1. `Service` in this context is not a very good variable name. It's too generic (in this context). You probably copied it from [here](https://github.com/saucelabs/saucectl/blob/abbff11694f40ed91183d23ba7df9e2dad102a21/internal/report/buildtable/buildtable.go#L20-L20). But in the context of a **build** table reporter, `Service` is more clear.
2. Considering that we'd now have to two places where the report has to be enriched with a build URL, would it make more sense to do refactor first so that build information is fetched [at the result collection time](https://github.com/saucelabs/saucectl/blob/abbff11694f40ed91183d23ba7df9e2dad102a21/internal/saucecloud/cloud.go#L99-L99? That way, we only have to make these calls once and the reporter(s) will have less side effects.